From: WANG Cong <xiyou.wangcong@gmail.com>
Date: Thu, 12 Jan 2012 17:20:11 -0800
Subject: kexec: remove KMSG_DUMP_KEXEC

commit a3dd3323058d281abd584b15ad4c5b65064d7a61 upstream.

KMSG_DUMP_KEXEC is useless because we already save kernel messages inside
/proc/vmcore, and it is unsafe to allow modules to do other stuffs in a
crash dump scenario.

[akpm@linux-foundation.org: fix powerpc build]
Signed-off-by: WANG Cong <xiyou.wangcong@gmail.com>
Reported-by: Vivek Goyal <vgoyal@redhat.com>
Acked-by: Vivek Goyal <vgoyal@redhat.com>
Acked-by: Jarod Wilson <jarod@redhat.com>
Cc: "Eric W. Biederman" <ebiederm@xmission.com>
Cc: KOSAKI Motohiro <kosaki.motohiro@jp.fujitsu.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---
 arch/powerpc/platforms/pseries/nvram.c |    1 -
 drivers/char/ramoops.c                 |    3 +--
 drivers/mtd/mtdoops.c                  |    3 +--
 include/linux/kmsg_dump.h              |    1 -
 kernel/kexec.c                         |    3 ---
 5 files changed, 2 insertions(+), 9 deletions(-)

Index: linux-3.2.46/arch/powerpc/platforms/pseries/nvram.c
===================================================================
--- linux-3.2.46.orig/arch/powerpc/platforms/pseries/nvram.c	2013-07-26 18:52:22.000000000 +0000
+++ linux-3.2.46/arch/powerpc/platforms/pseries/nvram.c	2013-07-26 19:30:27.000000000 +0000
@@ -636,7 +636,6 @@
 		/* These are almost always orderly shutdowns. */
 		return;
 	case KMSG_DUMP_OOPS:
-	case KMSG_DUMP_KEXEC:
 		break;
 	case KMSG_DUMP_PANIC:
 		panicking = true;
Index: linux-3.2.46/drivers/char/ramoops.c
===================================================================
--- linux-3.2.46.orig/drivers/char/ramoops.c	2013-07-26 18:52:22.000000000 +0000
+++ linux-3.2.46/drivers/char/ramoops.c	2013-07-26 19:30:27.000000000 +0000
@@ -83,8 +83,7 @@
 	struct timeval timestamp;
 
 	if (reason != KMSG_DUMP_OOPS &&
-	    reason != KMSG_DUMP_PANIC &&
-	    reason != KMSG_DUMP_KEXEC)
+	    reason != KMSG_DUMP_PANIC)
 		return;
 
 	/* Only dump oopses if dump_oops is set */
Index: linux-3.2.46/drivers/mtd/mtdoops.c
===================================================================
--- linux-3.2.46.orig/drivers/mtd/mtdoops.c	2013-07-26 18:52:22.000000000 +0000
+++ linux-3.2.46/drivers/mtd/mtdoops.c	2013-07-26 19:30:27.000000000 +0000
@@ -311,8 +311,7 @@
 	char *dst;
 
 	if (reason != KMSG_DUMP_OOPS &&
-	    reason != KMSG_DUMP_PANIC &&
-	    reason != KMSG_DUMP_KEXEC)
+	    reason != KMSG_DUMP_PANIC)
 		return;
 
 	/* Only dump oopses if dump_oops is set */
Index: linux-3.2.46/include/linux/kmsg_dump.h
===================================================================
--- linux-3.2.46.orig/include/linux/kmsg_dump.h	2013-07-26 18:52:22.000000000 +0000
+++ linux-3.2.46/include/linux/kmsg_dump.h	2013-07-26 19:30:27.000000000 +0000
@@ -18,7 +18,6 @@
 enum kmsg_dump_reason {
 	KMSG_DUMP_OOPS,
 	KMSG_DUMP_PANIC,
-	KMSG_DUMP_KEXEC,
 	KMSG_DUMP_RESTART,
 	KMSG_DUMP_HALT,
 	KMSG_DUMP_POWEROFF,
Index: linux-3.2.46/kernel/kexec.c
===================================================================
--- linux-3.2.46.orig/kernel/kexec.c	2013-07-26 18:52:22.000000000 +0000
+++ linux-3.2.46/kernel/kexec.c	2013-07-26 19:30:27.000000000 +0000
@@ -32,7 +32,6 @@
 #include <linux/console.h>
 #include <linux/vmalloc.h>
 #include <linux/swap.h>
-#include <linux/kmsg_dump.h>
 #include <linux/syscore_ops.h>
 
 #include <asm/page.h>
@@ -1094,8 +1093,6 @@
 		if (kexec_crash_image) {
 			struct pt_regs fixed_regs;
 
-			kmsg_dump(KMSG_DUMP_KEXEC);
-
 			crash_setup_regs(&fixed_regs, regs);
 			crash_save_vmcoreinfo();
 			machine_crash_shutdown(&fixed_regs);
